### PR TITLE
18 us 9 merchant item disableenable

### PR DESCRIPTION
**Pull request by megroth2**
It's not my best work, but I think I got it working... I ended up creating two tables - one for enabled items and one for disabled. This was so that I could ensure the button I'm pressing is updating the correct item.

- Added update action to items controller
- Added two helper methods in the merchant model
- Added tests for each helper method in the merchant model spec
- Added two tables with enable/disable buttons to the items index view
- Added a test for the tables in the items index view spec
- Added a status column to the items table
- Updated the item factory so it will default item status to disabled

### CODE REVIEW

File: spec/factories/item.rb
Comment (s2an) [inline spec/factories/item.rb:1]: I did this different. I left status as blank to denote disabled. I felt that maybe Active and Inactive would be better, but add an extra layer. Have a list of names labeled "disabled" could be misconstrued. Potato, tomato

File: db/migrate/20240110064616_add_status_to_items.rb
Comment (s2an) [inline db/migrate/20240110064616_add_status_to_items.rb:1]: Does this add three new columns?

File: config/routes.rb
Comment (s2an) [inline config/routes.rb:1]: I did not know you could put the params there.

File: spec/features/items/index_spec.rb [hunk 1]
```diff
@@ -30,4 +30,33 @@
...
+    it "displays enable and disable buttons to update the item status" do
+      visit "/merchants/#{@merchant_1.id}/items"
+  
+      within("tr:contains('#{@item_1.name}')") do
+        expect(page).to have_button("Enable")
+
+        click_button("Enable")
+
+        expect(current_path).to eq("/merchants/#{@merchant_1.id}/items")
+
```
Comment (s2an) [inline spec/features/items/index_spec.rb:51]: these could be eq(merchant_items_path(@item.merchant_id)

File: db/migrate/20240110064616_add_status_to_items.rb
Comment (megroth2) [reply db/migrate/20240110064616_add_status_to_items.rb:1]: nope, just one :status column on the :items table that is a :string!

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -15,6 +15,12 @@ def index
     @merchant = Merchant.find(params[:merchant_id])
   end
 
+  def update
+    @item = Item.find(params[:item_id])
+    @item.update(status: params[:status]) if params[:status].present?
+    redirect_to "/merchants/#{@item.merchant_id}/items"
+  end
```
[the remark this answers is not in the document]
Comment (megroth2) [reply app/controllers/items_controller.rb:22]: Ah yes, I haven't gotten around to updating my paths to use path helpers yet. Will do if there's time!

File: spec/factories/item.rb
Comment (megroth2) [reply spec/factories/item.rb:1]: Aaah see I thought I read a requirement that the default value should be disabled but I must've made it up!

**Pull request merged**